optaplanner-core/src/main/java/org/optaplanner/core/api/score/ScoreManager.java

```diff
@@ -61,4 +63,33 @@
      */
     void updateScore(Solution_ solution);
 
+    /**
+     * Returns the {@link Score} of the solution.
+     * <p>
+     * This is useful for generic code, which doesn't know the type of the {@link PlanningSolution} to retrieve the
+     * {@link Score} from the solution easily.
+     * <p>
+     * This method is thread-safe.
+     *
+     * @return null if the {@link PlanningSolution} is still uninitialized
+     */
+    Score getScore(Solution_ solution);
+
+    /**
+     * Returns a diagnostic text that explains the solution through the {@link ConstraintMatch} API to identify which
+     * constraints or planning entities cause that score quality.
+     * In case of an {@link Score#isFeasible() infeasible} solution, this can help diagnose the cause of that.
+     * <p>
+     * Do not parse this string.
+     * Instead, to provide this information in a UI or a service, use {@link SolverFactory#getScoreDirectorFactory()}
+     * to retrieve {@link ScoreDirector#getConstraintMatchTotalMap()} and {@link ScoreDirector#getIndictmentMap()}
+     * and convert those into a domain specific API.
+     * <p>
+     * This method is thread-safe.
+     *
+     * @return null if {@link #getScore(Object)} returns null with the same solution
+     * @see ScoreDirector#explainScore()
+     */
+    String explainScore(Solution_ solution);
```

Comment:
New API. Technically, this is a break of bwd compatibility for the interface.
Partically, no way to avoid it.